pkg/vpcmodel/grouping.go

```diff
@@ -643,22 +643,28 @@ func listEndpointElemStrWithConfig(eps []EndpointElem, fn func(ep EndpointElem,
 }
 
 func (g *groupedExternalNodes) String() string {
+	// 1 gets list and a union of all IPBlocks
+	unionBlock := g.toIPBlocks()
+	// 2. print a list s.t. each element contains either a single cidr or an ip range
+	return strings.Join(unionBlock.ListToPrint(), commaSeparator)
+}
+
+func (g *groupedExternalNodes) toIPBlocks() *ipblock.IPBlock {
 	// 1. Created a list of IPBlocks
 	cidrList := make([]string, len(*g))
 	for i, n := range *g {
 		cidrList[i] = n.CidrStr
 	}
 	ipbList, _, err := ipStringsToIPblocks(cidrList)
 	if err != nil {
-		return ""
+		return nil
```

Comment:
I think it's better to return an empty IPBlock, to avoid nil dereferencing issues

Reply:
ok, done